recipes/lora_finetune_single_device.py

```diff
@@ -596,14 +611,20 @@ def train(self) -> None:
                         self.global_step += 1
 
                         loss_to_log = running_loss.item()
+                        rounded_loss = round(loss_to_log, 5)
                         pbar.update(1)
                         pbar.set_description(
-                            f"{curr_epoch + 1}|{self.global_step}|Loss: {loss_to_log}"
+                            f"{curr_epoch + 1}|{self.global_step}|{self.color.green}Loss: {rounded_loss} "
+                            + f"{self.color.white}| {self.color.blue}TPS: {self.tps} {self.color.white}|"
+                            + f"{self.color.yellow} Mem: {self.mem_pct:.2f}% "
+                            + f"{self.color.white}| {self.color.red} Retries: {self.mem_cache_retries}{self.color.reset}"
                         )
 
                         # Log per-step metrics
                         if self.global_step % self._log_every_n_steps == 0:
                             time_per_step = time.perf_counter() - t0
+                            self.tps = round(num_tokens / time_per_step, 3)
```

Comment:
this is calculated in line 631. We should probably have only one reference for it. Maybe update line 631?

nit: toks_per_sec should be more friendly, if you dont think its too long

Reply:
yes on both points.